008_team_6_agent_design/008_team_6_agent_design.tex

```diff
@@ -1 +1,709 @@
+        default:
+            return a.foodRequired(levels.healthyLevel, healthInfo
+    }
+\end{verbatim}
+
+\subsubsection{Narcissistic}
+A narcissistic agent will always return the maximum amount of food consumable:
+
+\begin{verbatim}
+    case "Narcissist": // Eat max intake
+        return healthInfo.maxIntake
+\end{verbatim}
+
+\newpage
+
+%%%%%%%%%%%
+%%%%%%%%%%% SELF-ORGANIZATION
+%%%%%%%%%%%
+\section{Self-Organization}\label{self-organization}
+
+“Self-organizing” makes up almost half of this course's name. Accordingly, we approach self-organization capabilities as fundamental to our agents behaviour.
+
+Messages allow for short-lived communication between two neighbours. With request and response messages, such as “leave X food for me”, agents can organize locally. However, to establish a stable, self-organizing system across many more floors and many reassignment periods, a more sophisticated form of agreement is necessary. For that reason, an agent's strategies relies heavily on treaties, mimicking research findings from the field of behavioural psychology.
+
+To successfully handle treaties, an agent must:
+
+\begin{enumerate}
+    \item Rate treaties
+    \item Change its behaviour based on accepted treaties
+    \item Propose treaties
+\end{enumerate}
+
+Before we explain the theory and implementation answering each question, it is necessary to narrow down what types of treaties could lead to stable systems. 
+
+\subsection{Types of treaties}
+
+As food is a scarce good in the tower, the main purpose of self-organization should be to limit how much food an agent eats. While this strategy handles all possible conditions, (”less than”, ”less than or equal”, “equal to”, “more than or equal to”, “more than”), it focuses on upper bounds to the amount of food we eat. Table X show how different social motives would handle each form of treaty.
+
+\ToDo{Table with motives}
+
+The infrastructure of the tower provides multiple types of treaties; “leaveAmountFood”, “leavePercentageFood” and “inform”. \ToDo{It wasn't clear to me from first read that the next sentences were discussing the problems of each of the treaty types, maybe have to make it more clear?} “leaveAmountFood” would allow the first few floors to eat as much as it wants without breaking the treaty. Once the amount is reached, any subsequent agent does not receiving anything. Similarly, “leavePercentageFood” limits agents by exponentially decreasing portions, such that low levels have little chance of surviving even if everyone satisfies the treaty. "takeAmountFood" does not face this issue as it limits consumption independently of the amount of food an agent receives. Consequently, the agent thinks in terms of the amount food it takes, converting any other treaty type to an equivalent upper bound of food it is allowed take. This makes it easier to compare treaties and make sure that treaties are not contradicting one another.
```

Comment:
The quotation marks are bugged. Consider using `` " or we could use csquotes package